src/main/java/net/glowstone/io/anvil/RegionFile.java

```diff
@@ -200,19 +214,19 @@ public DataInputStream getChunkDataInputStream(int x, int z) throws IOException
         if (version == VERSION_GZIP) {
             byte[] data = new byte[length - 1];
             file.read(data);
-            return new DataInputStream(new GZIPInputStream(new ByteArrayInputStream(data)));
+            return new DataInputStream(new BufferedInputStream(new GZIPInputStream(new ByteArrayInputStream(data))));
```

Comment:
The scope of this PR is purely to improve `RegionFile`'s memory and CPU consumption and increase its performance, so changing serialization is off the table.